tutorials/basic-topics/sorting/cyclic-sort.md

```diff
@@ -11,4 +11,97 @@ keywords:
   - cyclic sort
 ---
 
-<TutorialAuthors names="@TBC"/>
+<TutorialAuthors names="@prishit55"/>
+
+# Overview
+
+Cycle-Sort is an in-place, unstable-sorting algorithm.
+* `in-place` : An in-place algorithm transforms input without using any other auxiliary data structure. As the algorithm executes the input is overwritten by the output.
+* `unstable` : Unstable sorting algorithm don't preserve the relative order of equal elements while sorting.
+
+
+Cycle sort algorithm factors the permutation to be sorted into number of cycles, which are individually rotated to give desired sorted result.
+
+## Algorithm
+
+Consider an array with `n` distinct elements. For any element `x` we can find the index at which it will occur in the sorted array by counting the number of elements smaller than `x` in the entire array.Now,
+* if the element is at the correct position then do nothing 
+* Else, write the element to its intended position. That position must be inhabited by a different number `y` which has to be moved to its correct position. The process continues until we get an element at the original position of `x`.
+
+This completes one cycle. Repeating the cycle for every element will generate a sorted array.
+
+## C++ Implementation
+
+For a better understanding, let us take up a leetcode problem 
+
+Problem : 442. Find All Duplicates in an Array
+
+Link : https://leetcode.com/problems/find-all-duplicates-in-an-array/
+
+Problem Statement : An array of integers in the range [1,n] is given, each integer appears once or twice. We have to find all the integers that appear twice in the array.
+
+*Naive Approach* would be to simply use a map or a frequency array to store the frequency of each element and return an array of elements appearing twice. But we require constant space complexity.
+
+*Efficient Approach* is to put each element at its each correct index in the array as we have all integers in the range [1,n], and check if any of the element is not at its correct position then it is a duplicate element. Hence using `Cycle Sort` algorithm for this problem.
+```cpp
+vector<int> findDuplicates(vector<int>& nums) {
+        vector<int> duplicates;
+        int n = nums.size();
+        
+        // cyclc sort
+        int i = 0;
+        while(i < n) {
+            int index = nums[i] - 1;       //correct index
+            if(nums[i] != nums[index]) {   // if correct element is not present at the index
+                swap(nums[i], nums[index]); // we swap the elements
+            } else {
+                i++;     //do nothing if element is present at its correct position
+            }
+        }
+        
+        for(int i = 0; i < n; i++) {
+            if(nums[i] != i + 1) {              //extract all those elements which are not present 
+                duplicates.push_back(nums[i]); //at their correct position
+            }
+        }
+        return duplicates;
+    }
+```
```

Comment:
1. use the following format

````
<Tabs>
<TabItem value="cpp" label="C++">
<SolutionAuthor name="@yourname"/>

```cpp
// your code goes here
```

</TabItem>
</Tabs>
````

2. don't put inline comments. put it above the line.

3. include full code. not just the function. i.g.

```
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        // ...
    }
};
```

Reply:
Corrected!